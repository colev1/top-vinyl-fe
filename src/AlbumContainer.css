.albums-container {
  display: flex;
  flex-wrap: wrap;
}

.album {
  width: 300px;
}

h1:hover {
  color: red;
  cursor: pointer;
}