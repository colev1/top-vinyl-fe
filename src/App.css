.App {
  text-align: center;
  font-family: 'Inconsolata', monospace;
}

img {
  height: 80px;
  margin-right: 20px;
}

.title {
  /* height: 100px; */
  font-size: 44px;
  display: flex;
  justify-content: center;
  align-items: center;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
