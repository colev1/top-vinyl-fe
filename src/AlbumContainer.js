import React, {Component} from 'react';
import Album from './Album'
import './AlbumContainer.css';

class AlbumContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      displayTracks: false,
      albumId: null
    }
  }

  displayAlbumInfo = (id) => {
    console.log(id)
    this.setState({
      displayTracks: true,
      albumId: id
    })
  }
  
  render () {
  let albumsDisplay = this.props.albums.map(album => {
    return <div key={album.id} 
    className='album' 
    onClick={() => this.displayAlbumInfo(album.id)}>
      <h1> {album.album} </h1>
      <h3> {album.artist} </h3>
      <p> {album.genre} </p>
    </div>
    }
  )

  if(this.state.displayTracks) {
   return (
      <Album albumId={this.state.albumId}/>
   ) 
  } else {
    return (
      <div className='albums-container'>
        {albumsDisplay}
      </div>
    )
    }
  }
}

export default AlbumContainer;