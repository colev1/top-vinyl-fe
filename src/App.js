import React, { Component } from 'react';
import AlbumContainer from './AlbumContainer'
import './App.css';

class App extends Component {
  constructor () {
    super()
    this.state = {
      albums: [],
      error: '',
      album: '',
      genre: '',
      rating: '',
      artist: '',
      year: ''
    }
  }

  componentDidMount = () => {
    this.fetchAlbums()
  }

  fetchAlbums = () => {
    fetch('http://localhost:3000/api/v1/albums')
      .then(response => response.json())
      .then(result => this.addAlbums(result))
      .catch(error => console.log(error))
  }

  addAlbums = (albums) => {
    this.setState({
      albums
    })
  }

  deleteAlbum = (id) => {
    fetch(`http://localhost:3000/api/v1/albums/${id}`, {
      method: 'DELETE'
    })
    .then(response => response.json())
    .then(result => this.fetchAlbums())
    .catch(error => console.log(error))
  }

  handleChange = (e) => {
    const {name, value} = e.target;
    this.setState({
      [name]: value
    })
  }

  handleSubmit = (e) => {
    e.preventDefault()
    const {artist, genre, year, album} = this.state;
    const rating = parseInt(this.state.rating)
    const requestBody = {artist, genre, year, rating, album};
    fetch('http://localhost:3000/api/v1/albums', {
      method: 'POST',
      body: JSON.stringify(requestBody),
      headers: {
        "Content-Type": "application/json"
      }
    })
    .then(response => response.json())
    .then(result => console.log(result))
    .then(this.fetchAlbums())
    .catch(error => console.log(error))
  }

  render() {
    let {album, artist, year, genre, rating} = this.state;
    return (
      <div className="App">
        <h1> TOP VINYL </h1>

        <form onSubmit={(e) => this.handleSubmit(e)}> 
          add a new album:
          <input placeholder='album name' 
          name="album"
          value={album}
          onChange={this.handleChange} />
          <input placeholder='artist'
          name="artist"
          value={artist}
          onChange={this.handleChange} />
          <input placeholder='genre'
          name="genre"
          value={genre}
          onChange={this.handleChange} />
          <input placeholder='year'
          name="year"
          value={year}
          onChange={this.handleChange} />
          <input placeholder='rating'
          name="rating"
          value={rating}
          onChange={this.handleChange} />
          <button type="submit"
          > 
          add new album </button>
        </form>
        <AlbumContainer albums={this.state.albums} deleteAlbum={this.deleteAlbum}/>
      </div>
    );
  }
}

export default App;
