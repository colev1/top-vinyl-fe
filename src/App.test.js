import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';
import AlbumContainer from './AlbumContainer.js'

describe('App', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('should match the snapshot', () => {
    
    expect(wrapper).toMatchSnapshot()
  })

  it('should render an album container', () => {

    expect(wrapper.find(AlbumContainer).length).toEqual(1)
  })

  it('should call routeHome and change the state if the header is clicked', () => {
    const spy = jest.spyOn(wrapper.instance(), 'routeHome')
    wrapper.instance().forceUpdate()

    wrapper.find('.title').simulate('click')

    expect(spy).toHaveBeenCalled()
    expect(wrapper.state().displayTracks).toBe(false)
  })

  it('should fire handleChange on change of the album input and update the state', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleChange')
    const event = {
      target: {
        name: 'album',
        value: 'New Album'
      }
    }
    const expected = 'New Album'
    wrapper.instance().forceUpdate()

    wrapper.find('.album-input').simulate('change', event)

    expect(spy).toHaveBeenCalled()
    expect(wrapper.state().album).toEqual(expected)
  })

  it('should fire handleChange on change of the artist input and update the state', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleChange')
    const event = {
      target: {
        name: 'artist',
        value: 'Beyonce'
      }
    }
    const expected = 'Beyonce'
    wrapper.instance().forceUpdate()

    wrapper.find('.artist-input').simulate('change', event)

    expect(spy).toHaveBeenCalled()
    expect(wrapper.state().artist).toEqual(expected)
  })

  it('should fire handleChange on change of the genre input and update state', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleChange')
    const event = {
      target: {
        name: 'genre',
        value: 'Pop'
      }
    }
    const expected = 'Pop'
    wrapper.instance().forceUpdate()

    wrapper.find('.genre-input').simulate('change', event)

    expect(spy).toHaveBeenCalled()
    expect(wrapper.state().genre).toEqual(expected)
  })

  it('should fire handleChange on change of the year input and update state', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleChange')
    const event = {
      target: {
        name: 'year',
        value: '2011'
      }
    }
    const expected = '2011'
    wrapper.instance().forceUpdate()

    wrapper.find('.year-input').simulate('change', event)

    expect(spy).toHaveBeenCalled()
    expect(wrapper.state().year).toEqual(expected)
  })

  it('should fire handleChange on change od the rating input and update state', () => {
    const spy = jest.spyOn(wrapper.instance(), 'handleChange')
    const event = {
      target: {
        name: 'rating',
        value: 3.24
      }
    }
    const expected = '3.24'
    wrapper.instance().forceUpdate()

    wrapper.find('.rating-input').simulate('change', event)

    expect(spy).toHaveBeenCalled()
    expect(wrapper.state().rating).toEqual(expected)
  })
})