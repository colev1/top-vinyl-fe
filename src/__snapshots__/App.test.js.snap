// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1
            className="title"
            onClick={[Function]}
          >
             
            <img
              alt="Vinyl"
              src="vinyl.png"
            />
             TOP VINYL 
          </h1>
          add a new album:
          <form
            onSubmit={[Function]}
          >
            <input
              className="album-input"
              name="album"
              onChange={[Function]}
              placeholder="album name"
              value=""
            />
            <input
              className="artist-input"
              name="artist"
              onChange={[Function]}
              placeholder="artist"
              value=""
            />
            <input
              className="genre-input"
              name="genre"
              onChange={[Function]}
              placeholder="genre"
              value=""
            />
            <input
              className="year-input"
              name="year"
              onChange={[Function]}
              placeholder="year"
              value=""
            />
            <input
              className="rating-input"
              name="rating"
              onChange={[Function]}
              placeholder="rating 0-5"
              value=""
            />
            <button
              type="submit"
            >
              add new album 
            </button>
          </form>
        </header>,
        <AlbumContainer
          albums={Array []}
          deleteAlbum={[Function]}
          displayTracks={false}
          toggleDisplay={[Function]}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="title"
              onClick={[Function]}
            >
               
              <img
                alt="Vinyl"
                src="vinyl.png"
              />
               TOP VINYL 
            </h1>,
            "add a new album:",
            <form
              onSubmit={[Function]}
            >
              <input
                className="album-input"
                name="album"
                onChange={[Function]}
                placeholder="album name"
                value=""
              />
              <input
                className="artist-input"
                name="artist"
                onChange={[Function]}
                placeholder="artist"
                value=""
              />
              <input
                className="genre-input"
                name="genre"
                onChange={[Function]}
                placeholder="genre"
                value=""
              />
              <input
                className="year-input"
                name="year"
                onChange={[Function]}
                placeholder="year"
                value=""
              />
              <input
                className="rating-input"
                name="rating"
                onChange={[Function]}
                placeholder="rating 0-5"
                value=""
              />
              <button
                type="submit"
              >
                add new album 
              </button>
            </form>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                <img
                  alt="Vinyl"
                  src="vinyl.png"
                />,
                " TOP VINYL ",
              ],
              "className": "title",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Vinyl",
                  "src": "vinyl.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              " TOP VINYL ",
            ],
            "type": "h1",
          },
          "add a new album:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="album-input"
                  name="album"
                  onChange={[Function]}
                  placeholder="album name"
                  value=""
                />,
                <input
                  className="artist-input"
                  name="artist"
                  onChange={[Function]}
                  placeholder="artist"
                  value=""
                />,
                <input
                  className="genre-input"
                  name="genre"
                  onChange={[Function]}
                  placeholder="genre"
                  value=""
                />,
                <input
                  className="year-input"
                  name="year"
                  onChange={[Function]}
                  placeholder="year"
                  value=""
                />,
                <input
                  className="rating-input"
                  name="rating"
                  onChange={[Function]}
                  placeholder="rating 0-5"
                  value=""
                />,
                <button
                  type="submit"
                >
                  add new album 
                </button>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "album-input",
                  "name": "album",
                  "onChange": [Function],
                  "placeholder": "album name",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "artist-input",
                  "name": "artist",
                  "onChange": [Function],
                  "placeholder": "artist",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "genre-input",
                  "name": "genre",
                  "onChange": [Function],
                  "placeholder": "genre",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "year-input",
                  "name": "year",
                  "onChange": [Function],
                  "placeholder": "year",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "rating-input",
                  "name": "rating",
                  "onChange": [Function],
                  "placeholder": "rating 0-5",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "add new album ",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "add new album ",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "albums": Array [],
          "deleteAlbum": [Function],
          "displayTracks": false,
          "toggleDisplay": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1
              className="title"
              onClick={[Function]}
            >
               
              <img
                alt="Vinyl"
                src="vinyl.png"
              />
               TOP VINYL 
            </h1>
            add a new album:
            <form
              onSubmit={[Function]}
            >
              <input
                className="album-input"
                name="album"
                onChange={[Function]}
                placeholder="album name"
                value=""
              />
              <input
                className="artist-input"
                name="artist"
                onChange={[Function]}
                placeholder="artist"
                value=""
              />
              <input
                className="genre-input"
                name="genre"
                onChange={[Function]}
                placeholder="genre"
                value=""
              />
              <input
                className="year-input"
                name="year"
                onChange={[Function]}
                placeholder="year"
                value=""
              />
              <input
                className="rating-input"
                name="rating"
                onChange={[Function]}
                placeholder="rating 0-5"
                value=""
              />
              <button
                type="submit"
              >
                add new album 
              </button>
            </form>
          </header>,
          <AlbumContainer
            albums={Array []}
            deleteAlbum={[Function]}
            displayTracks={false}
            toggleDisplay={[Function]}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="title"
                onClick={[Function]}
              >
                 
                <img
                  alt="Vinyl"
                  src="vinyl.png"
                />
                 TOP VINYL 
              </h1>,
              "add a new album:",
              <form
                onSubmit={[Function]}
              >
                <input
                  className="album-input"
                  name="album"
                  onChange={[Function]}
                  placeholder="album name"
                  value=""
                />
                <input
                  className="artist-input"
                  name="artist"
                  onChange={[Function]}
                  placeholder="artist"
                  value=""
                />
                <input
                  className="genre-input"
                  name="genre"
                  onChange={[Function]}
                  placeholder="genre"
                  value=""
                />
                <input
                  className="year-input"
                  name="year"
                  onChange={[Function]}
                  placeholder="year"
                  value=""
                />
                <input
                  className="rating-input"
                  name="rating"
                  onChange={[Function]}
                  placeholder="rating 0-5"
                  value=""
                />
                <button
                  type="submit"
                >
                  add new album 
                </button>
              </form>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <img
                    alt="Vinyl"
                    src="vinyl.png"
                  />,
                  " TOP VINYL ",
                ],
                "className": "title",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Vinyl",
                    "src": "vinyl.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                " TOP VINYL ",
              ],
              "type": "h1",
            },
            "add a new album:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="album-input"
                    name="album"
                    onChange={[Function]}
                    placeholder="album name"
                    value=""
                  />,
                  <input
                    className="artist-input"
                    name="artist"
                    onChange={[Function]}
                    placeholder="artist"
                    value=""
                  />,
                  <input
                    className="genre-input"
                    name="genre"
                    onChange={[Function]}
                    placeholder="genre"
                    value=""
                  />,
                  <input
                    className="year-input"
                    name="year"
                    onChange={[Function]}
                    placeholder="year"
                    value=""
                  />,
                  <input
                    className="rating-input"
                    name="rating"
                    onChange={[Function]}
                    placeholder="rating 0-5"
                    value=""
                  />,
                  <button
                    type="submit"
                  >
                    add new album 
                  </button>,
                ],
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "album-input",
                    "name": "album",
                    "onChange": [Function],
                    "placeholder": "album name",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "artist-input",
                    "name": "artist",
                    "onChange": [Function],
                    "placeholder": "artist",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "genre-input",
                    "name": "genre",
                    "onChange": [Function],
                    "placeholder": "genre",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "year-input",
                    "name": "year",
                    "onChange": [Function],
                    "placeholder": "year",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "rating-input",
                    "name": "rating",
                    "onChange": [Function],
                    "placeholder": "rating 0-5",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "add new album ",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "add new album ",
                  "type": "button",
                },
              ],
              "type": "form",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "albums": Array [],
            "deleteAlbum": [Function],
            "displayTracks": false,
            "toggleDisplay": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
